test/tests_for_sdkmr/EE2Runjob_test.py

```diff
@@ -56,8 +62,8 @@
 _USER = "someuser"
 _TOKEN = "tokentokentoken"
 _OTHER_USER = "some_sucker"
-_CREATED_STATE = "created"
-_QUEUED_STATE = "queued"
+_CREATED_STATE = Status.created.value
+_QUEUED_STATE = Status.queued.value
```

Comment:
I thought it was weird to think of checking these strings directly. In that case, doesn't that mean we should also be checking to make sure  that the strings we are hardcoding in here  pass the validity checking functions in models.py like `valid_status` `valid_authstrat` `valid_termination_code` and `valid_errorcode`  ?

Reply:
We do that in the `api_to_db_test`, right?

Reply:
this is a unit test just for EE2RunJob, so I don't think so. It really shouldn't care about how other modules handle the strings - just that the strings are part of the expected output from the method calls

Reply:
The source of truth for these values should be from the Status enum, so not using it seems very wrong to me. But ok, will change it back.

Reply:
> following this logic we can make these strings to be "cat" or "dog" within the test if we'd like

If we did that the tests would fail right? The states are set in `_init_job_rec` and `update_job_to_queued`. When the `save_job.assert_called_once_with` or the equivalent for update is called in the tests, those would fail if the state strings are wrong.

That's the point of not using constructs from the code under test in the test - if those constructs change then the tests fail, which is what we want. 

So the source of truth is still the `Status` values and these tests ensure the the source of truth is not changed without being a deliberate action on the part of the coder.

Reply:
anyway, fixed so :+1:

Reply:
So why would we use these tests to ensure that the source of truth is not changed, rather than a test dedicated to checking the source of truth?

Reply:
You could certainly have both. These tests ensure that the source of truth is used correctly in EE2RunJob though, which a test that only checks the contents of the source of truth wouldn't do.